src/differentiation/compute_jacobian_ad.jl

```diff
@@ -79,23 +93,37 @@ function forwarddiff_color_jacobian!(J::AbstractMatrix{<:Number},
     p = jac_cache.p
     color = jac_cache.color
 
-    # TODO: Should compute on each p[i] and decompress
-    t .= Dual{typeof(f)}.(x, p)
-    f(fx, t)
-
-    if J isa SparseMatrixCSC
-        rows_index, cols_index, val = findnz(J)
-        for color_i in 1:maximum(color)
-            dx .= partials.(fx,color_i)
-            for i in 1:length(cols_index)
-                if color[cols_index[i]]==color_i
-                    J[rows_index[i],cols_index[i]] = dx[rows_index[i]]
+    color_i = 1
+    chunksize = getsize(default_chunk_size(maximum(color)))
+
+    for i in 1:length(p)
+        partial_i = p[i]
+        t .= Dual{typeof(f)}.(x, partial_i)
+        f(fx,t)
+
+        if J isa SparseMatrixCSC
+            rows_index, cols_index, val = findnz(J)
+            for j in 1:chunksize
+                dx .= partials.(fx, j)
+                for k in 1:length(cols_index)
+                    if color[cols_index[k]] == color_i
+                        J[rows_index[k], cols_index[k]] = dx[rows_index[k]]
+                    end
+                end
+                color_i += 1
+                if color_i == maximum(color) + 1
+                    color_i = 1
+                end
+            end
+        else
+            for j in 1:chunksize
+                col_index = (i-1)*chunksize + j
+                J[:, col_index] .= partials(fx, color_i)
+                color_i += 1
+                if color_i == maximum(color) + 1
```

Comment:
this will only happen at the end?

Reply:
Oh I see what this is. The name is confusing because it's not longer necessarily the color. Name it like `cur_partial`. And shouldn't it reset after it hits the chunksize? Otherwise `partials(fx, color_i)` can have `color_i > chunksize` and it will index a partial that doesn't exist

Reply:
Yeah you're right. But I think we can just remove the `color_i` variable entirely and simply copy the columns from the partials to the Jacobian using
~~~
for j in 1:chunksize
    col_index = (i-1)*chunksize + j
    J[:, col_index] .= partials.(fx, j)
end
~~~
We use `col_index` variable to map which column in the original matrix would the the `jth` column in the `ith` pass of the chunked partials would correspond to, using the equation supplied.